Add .env option to specify Cyclone DDS network interface settings, fix DDS setting service shell for multiple hosts

tatsuya-ishihara opened the pull request:
Instead of following .env setting,
```
CYCLONEDDS_URI=/home/developer/cyclonedds_spdp.xml
```

I added following .env settings to change Cyclone DDS multicast and network interface settings.
```
CYCLONEDDS_ALLOW_MULTICAST=spdp
CYCLONEDDS_NETWORK_INTERFACE_NAME=(network interface name, e.g. eth1)
```

`CYCLONEDDS_NETWORK_INTERFACE_NAME` is necessary when PC has multiple network interface.

DCO 1.1 Signed-off-by: Tatsuya Ishihara <tisihara@jp.ibm.com>

Commit:
Fix .env options to change Cyclone DDS multicast and network interface settings

DCO 1.1 Signed-off-by: Tatsuya Ishihara <tisihara@jp.ibm.com>

dsato80 commented on README.md:
```diff
@@ -135,7 +135,7 @@ CaBot v2 has been migrated to ROS2 (humble). It uses Docker containers to mainta
     #
     # ex) "D:192.168.1.50:rs1 D:192.168.1.51:rs2"
-  CYCLONEDDS_URI=/home/developer/cyclonedds_spdp.xml   # need to use cyclone dds SPDP message option for multiple hosts
+  CYCLONEDDS_ALLOW_MULTICAST=spdp   # need to use cyclone dds SPDP message option for multiple hosts
```
I think Jetson may use `CYCLONEDDS_URI` directory. `CYCLONEDDS_ALLOW_MULTICAST` is only used in `launch.sh` which is not used in Jetson.

dsato80 replied:
Is `CYCLONEDDS_ALLOW_MULTICAST` required as it is always `spdp`?

tatsuya-ishihara replied:
Jetson uses `CYCLONEDDS_URI` directly.
I will remove `CYCLONEDDS_ALLOW_MULTICAST` by preparing two XML files (with NetworkInterface, without NetworkInterface).

dsato80 commented on docker/home/cyclonedds_spdp.xml:
Is this for Jetson?

dsato80 replied:
how about preparing another xml file that has an Interface tag and switching `CYCLONEDDS_URI` based on the existence of `CYCLONEDDS_NETWORK_INTERFACE_NAME` ?

tatsuya-ishihara replied:
OK. I will add another xml file, and switch by `CYCLONEDDS_NETWORK_INTERFACE_NAME`.

dsato80 commented on launch.sh:
I prefer not to include XML content in the shell.

tatsuya-ishihara replied:
I will remove XML content in the shell by preparing two XML files.

dsato80 reviewed (changes requested):
Could you check my comments? Thank you!

Commit:
Rollback XML config file to change Cyclone DDS settings
Use autodetermine option to select interface, and add log option for DDS.

DCO 1.1 Signed-off-by: Tatsuya Ishihara <tisihara@jp.ibm.com>

Commit:
Change IP fragmentation settings in DDS setting shell for multiple hosts

DCO 1.1 Signed-off-by: Tatsuya Ishihara <tisihara@jp.ibm.com>

tatsuya-ishihara commented:
I found there is no need to create two Cyclone DDS XML files by using "autodetermine" attribute of Network Interface.
I think using one XML file will be better than switching two XML files.

To confirm Cyclone DDS settings later, I set option to save log file about Cyclone DDS configuration.

Also, I fixed "tools/change_dds_settings.sh" to fix the issue that rviz freezes when showing multiple images from remote host.

Please review the changes.

daisukes commented on docker-compose-common.yaml:
```diff
@@ -59,6 +59,8 @@ services:
       - RMW_IMPLEMENTATION
       - CYCLONEDDS_URI
+      - CYCLONEDDS_NETWORK_INTERFACE_NAME
+      - CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE
```
Do we need this variable `CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE` to be specified here? If yes, please describe it in the README. You specify the value in the launch script, so we may not need this here. It will be overridden anyway.

tatsuya-ishihara replied:
`CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE` is used when loading XML file specified by `CYCLONEDDS_URI` in docker, so we need to specify here.

When network interface name is set, `CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE` should be `false`.
When network interface name is not set, `CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE` should be `true`. 

Adding the variable which is set automatically may be confusing,  so I added following comment in docker-compose-common.xml
```
- CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE # set automatically by launch.sh
```

Do we need to write README about variable which will be automatically set?

daisukes commented on docker-compose-jetson.yaml:
```diff
@@ -35,6 +35,8 @@ services:
       GAZEBO_MASTER_URI: http://${MASTER_IP:-127.0.0.1}:11345
       CYCLONEDDS_URI: ${CYCLONEDDS_URI}
+      CYCLONEDDS_NETWORK_INTERFACE_NAME: ${CYCLONEDDS_NETWORK_INTERFACE_NAME}
+      CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE: ${CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE}
```
What happens if you do not specify `CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE` in Jetson? You may want to specify a default value.

tatsuya-ishihara replied:
The default value should be set. I updated the file.

daisukes reviewed (changes requested):
Thank you for the update.
Could you check my comments?

Commit:
Added comment about CYCLONEDDS_NETWORK_INTERFACE_AUTODETERMINE in docker-compose, and set default value for jetson docker-compose

DCO 1.1 Signed-off-by: Tatsuya Ishihara <tisihara@jp.ibm.com>

tatsuya-ishihara commented:
Thank you for your comment. I commit changes. Please let me know if I need to update README file too.

daisukes reviewed (approved):
LGTM
